src/main/kotlin/org/rust/cargo/runconfig/buildtool/RsBuildEventsConverter.kt

```diff
@@ -299,16 +300,20 @@ class RsBuildEventsConverter(private val context: CargoBuildContext) : BuildOutp
             }
 
         private fun createMessageEvent(
+            workingDirectory: Path,
             parentEventId: Any,
             kind: MessageEvent.Kind,
             message: String,
             detailedMessage: String?,
             filePosition: FilePosition? = null
-        ): MessageEvent = if (filePosition == null) {
-            MessageEventImpl(parentEventId, kind, RUSTC_MESSAGE_GROUP, message, detailedMessage)
-        } else {
-            FileMessageEventImpl(parentEventId, kind, RUSTC_MESSAGE_GROUP, message, detailedMessage, filePosition)
-        }
+        ): MessageEvent = FileMessageEventImpl(
+            parentEventId,
+            kind,
+            RUSTC_MESSAGE_GROUP,
+            message,
+            detailedMessage,
+            filePosition ?: FilePosition(workingDirectory.toFile(), 0, 0)
```

Comment:
Let's add a comment why we need this "fallback" file position and that we should use `MessageEventImpl` instead of `FileMessageEventImpl` when it will be fixed